src/metabase/upload.clj

```diff
@@ -176,11 +193,23 @@
 ;;;; +------------------+
 
 (defn unique-table-name
-  "Append the current datetime to the given name to create a unique table name."
-  [table-name]
-  (str (u/slugify table-name)
-       (t/format "_yyyyMMddHHmmss" (t/local-date-time))))
-
+  "Append the current datetime to the given name to create a unique table name. The resulting name will be short enough for the given driver (truncating the supplised `table-name` if necessary)."
+  [driver table-name]
+  (let [time-format                 "_yyyyMMddHHmmss"
+        acceptable-length           (min (count table-name)
+                                         (- (driver/table-name-length-limit driver) (count time-format)))
+        truncated-name-without-time (subs (u/slugify table-name) 0 acceptable-length)]
+    (str truncated-name-without-time
+         (t/format time-format (t/local-date-time)))))
+
+(def max-sample-rows "Maximum number of values to use for detecting a column's type" 1000)
+
+(defn- sample-rows
```

Comment:
I'm guessing this is trying to solve performance problems with `rows->schema`. The problem is the data in the first N rows might not match the rest of the data. We should try to optimize `rows->schema` before resorting to this.

Reply:
I just read the [convo](https://metaboat.slack.com/archives/C04S696LRUM/p1681920630385549?thread_ts=1681908673.932979&cid=C04S696LRUM) on slack, never mind. Makes sense to do this vs the status quo